### PR TITLE
Add and expand tests, policy JMESPath non-existence checks

### PR DESCRIPTION
**Pull request by chipzoller**
## Description

This PR adds and expands a number of tests in kyverno/policies. This does not cover the entirety of the repo, but does almost everything except the `other` folder. It also adds JMESPath non-existence checks to all applicable policies.
## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.

### CODE REVIEW

**Comment by chipzoller** [comment]
@samj1912 @JimBugwadia review is needed, please

File: cert-manager/limit-dnsnames/limit-dnsnames.yaml [hunk 1]
```diff
@@ -25,6 +25,6 @@ spec:
       message: Only one dnsNames entry allowed per certificate request.
       deny:
         conditions:
-        - key: "{{request.object.spec.dnsNames | length(@)}}"
+        - key: "{{request.object.spec.dnsNames || '' | length(@)}}"
```
Comment (samj1912) [inline cert-manager/limit-dnsnames/limit-dnsnames.yaml:28]:
```suggestion
        - key: "{{request.object.spec.dnsNames || `[]` | length(@)}}"
```

Comment (samj1912) [reply cert-manager/limit-dnsnames/limit-dnsnames.yaml:28]: Just to keep type consistency.

File: best-practices/disallow-empty-ingress-host/disallow_empty_ingress_host.yaml [hunk 2]
```diff
@@ -24,6 +24,6 @@ spec:
         message: "The Ingress host name must be defined, not empty."
         deny:
           conditions:
-            - key: "{{ request.object.spec.rules[].host | length(@) }}"
+            - key: "{{ request.object.spec.rules[].host || '' | length(@) }}"
```
Comment (samj1912) [inline best-practices/disallow-empty-ingress-host/disallow_empty_ingress_host.yaml:27]:
```suggestion
            - key: "{{ request.object.spec.rules[].host || `[]` | length(@) }}"
```

File: best-practices/disallow-empty-ingress-host/disallow_empty_ingress_host.yaml [hunk 3]
```diff
@@ -24,6 +24,6 @@ spec:
...
               operator: NotEquals
-              value: "{{ request.object.spec.rules[].http | length(@) }}"
\ No newline at end of file
+              value: "{{ request.object.spec.rules[].http || '' | length(@) }}"
```
Comment (samj1912) [inline best-practices/disallow-empty-ingress-host/disallow_empty_ingress_host.yaml:29]:
```suggestion
              value: "{{ request.object.spec.rules[].http || `[]` | length(@) }}"
```

**Review by samj1912 — COMMENTED** [review]
Some minor comments for type consistency when using default.

**Comment by chipzoller** [comment]
Addressed comments by fixing types.

**Review by samj1912 — APPROVED** [review]

**Pull request merged**